lib/service/fire_base.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:convert';
+
+import 'package:admin_dashboard/dto/model.dart';
+
+import 'basic_service.dart';
+import 'package:http/http.dart' as http;
+
+// final int issueID;
+// final String state;
+// final String title;
+// final DateTime logged_at;
+// final DateTime closed_at;
+// final String closed_by;
+// final int commentsNumber;
+// //final List<Assignee> assignees;
+// final String repository;
+class FireBaseService implements BasicServiceInterface{
+  @override
+  Future<Issue> addIssue(Issue issue) async {
+
+    DateTime now = DateTime.now();
+    String dateKey = ""+now.year.toString()+"_"+now.month.toString()+"_"+now.day.toString();
+    // date.toString().replaceAll("00:00:00.000", "");
+    // String yellow = date.toString();
+    const String repoNameTest= "plus_plugins";
+    const String issues= "issues";//eye yam steew peed
+    const String issueNumber = "908";
+    String siteUrl = 'https://admin-dashboard-b9503-default-rtdb.firebaseio.com/activity_log/' + repoNameTest +"/"+dateKey+"/"+ issues + '.json';
+    try {
+      final urlChat = Uri.parse(siteUrl);
+      Issue issue = Issue(250,"closed","Test Issue", DateTime.now(),DateTime.now(),"Amer",4,"Flutter Test" );
+      print(issue.toJson());
+      var response2 = await http.post(urlChat, body: json.encode(issue.toJson()));
+      print(response2.statusCode);
+      return issue;
+    } catch (error) {
+      rethrow;
+    }
```

Comment:
Well, while this is ok, it might bring up potential crash. it's better to define a custom Exception and handle it gracefully in the UI so use knows what's going on and why it's happening.